### PR TITLE
GH-36643: [C++][Parquet] Use nested namespace in parquet

### LINKED ISSUE #36643: [C++][Parquet] Use C++17's nested namespaces
### Describe the enhancement requested

Part of https://github.com/apache/arrow/issues/36620

Before:

```
namespace parquet {
namespace arrow {
```

After:

```
namespace parquet::arrow {
```

I guess most of these is under `parquet/arrow` directory.

### Component(s)

C++, Parquet

### PR DESCRIPTION
**Pull request by mapleFU**
### Rationale for this change

Just for code style, using nested namespace rather than before

### What changes are included in this PR?

Using nested namespace

### Are these changes tested?

no

### Are there any user-facing changes?

no

* Closes: #36643

### CODE REVIEW

**Review by wgtmac — APPROVED** [review]

File: cpp/src/parquet/level_conversion_inc.h [hunk 1]
```diff
@@ -30,8 +30,7 @@
 #include "parquet/exception.h"
 #include "parquet/level_comparison.h"
 
-namespace parquet {
-namespace internal {
+namespace parquet::internal {
 #ifndef PARQUET_IMPL_NAMESPACE
 #error "PARQUET_IMPL_NAMESPACE must be defined"
 #endif
```
Comment (kou) [inline cpp/src/parquet/level_conversion_inc.h:36]: We may want to use `namespace parquet::internal::PARQUET_IMPL_NAMESPACE`.

**Review by kou — APPROVED** [review]
+1

**Comment by mapleFU** [comment]
```
error    libmamba Failed to download package from https://conda.anaconda.org/conda-forge/nomkl-1.0-h5ca1d4c_0.tar.bz2 (status 404)
Multi-download failed. Reason: Transfer finalized, status: 404 [https://conda.anaconda.org/conda-forge/nomkl-1.0-h5ca1d4c_0.tar.bz2] 568 bytes
```

Seems this doesn't caused by my patch https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/47536246 ?

File: cpp/src/parquet/level_conversion_inc.h (hunk 1 above)
Comment (mapleFU) [reply cpp/src/parquet/level_conversion_inc.h:34]: Done

**Review by pitrou — APPROVED** [review]
A bit gratuitous, but LGTM

**Comment by pitrou** [comment]
> Seems this doesn't caused by my patch https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/47536246 ?

No, this build has been failing lately :-(

**Pull request merged**